frontend/src/pages/dataset/components/dataset-tabs/usage-tab/usage-chart.tsx

```diff
@@ -0,0 +1,144 @@
+import { Area } from '@ant-design/charts';
+import { Empty, Spin } from 'antd';
+import { format, subMonths } from 'date-fns';
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import type {
+    DatasetQueryStatsDailyResponse,
+    DatasetQueryStatsDailyResponses,
+} from '@/types/dataset/dataset-query-stats-daily.contract';
+
+type Props = {
+    data: DatasetQueryStatsDailyResponses | undefined;
+    isLoading: boolean;
+};
+
+type ChartDataPoint = {
+    date: string;
+    timestamp: number;
+    queryCount: number;
+    consumer: string;
+};
+
+function transformDataForChart(responses: DatasetQueryStatsDailyResponse[]): ChartDataPoint[] {
+    const oneMonthAgo = subMonths(new Date(), 1);
+    const now = new Date();
+
+    // transform date strings to Date objects
+    const processedData = responses
+        .map((stat) => {
+            const date = new Date(stat.date);
+            return {
+                timestamp: date.getTime(),
+                queryCount: stat.query_count,
+                consumer: stat.consumer_data_product_name || 'Unknown',
+            };
+        })
+        .filter((stat) => stat.timestamp >= oneMonthAgo.getTime())
+        .sort((a, b) => a.timestamp - b.timestamp);
+
+    // Get unique consumers
+    const consumers = [...new Set(processedData.map((d) => d.consumer))];
+
+    // Create a list of all dates in the range formatted as 'MMM d' or 'd'
+    const allDates: Array<{ date: string; timestamp: number }> = [];
+    let currentDate = new Date(oneMonthAgo);
+    let lastMonth = -1;
+    while (currentDate <= now) {
+        const currentMonth = currentDate.getMonth();
+        const dateLabel = currentMonth !== lastMonth ? format(currentDate, 'MMM d') : format(currentDate, 'd');
+        lastMonth = currentMonth;
+
+        allDates.push({
+            date: dateLabel,
+            timestamp: currentDate.getTime(),
+        });
+        currentDate = new Date(currentDate.getTime() + 24 * 60 * 60 * 1000); // Add one day
```

Comment:
You can use `date-fns` for this:

```suggestion
import { subDays } from 'date-fns';
        currentDate = subDays(currentDate, 1)
```

Reply:
now moved to `currentDate = addDays(currentDate, 1);`